icefoganalytics/sfa-client
# Messaging Module: Template Updates: Yukon Grant - STUDENT

Relates to:
- https://github.com/icefoganalytics/sfa-client/issues/3

# Context

Migrate Yukon Grant - STUDENT template from docx to the newer pdf and/or html format.

## Extras
- add support for html format for easier template testing.

## Notes

- Add original template letters package zip; the zip should be deleted once the new templates are finalized.

## Questions/Considerations

1. The `.pdf` is multiple pages, while original letter was only a single page. The original formatting was pretty questionable, but the "single page" letter might have been a requirement?
2. We might want to also support having the endpoint return the `.docx` template file for QA purposes?
3. We might want to investigate using Tailwind for back-end CSS?
4. We might want to investigate a better pattern for having letter specific CSS.

# Testing Instructions

1. Check that the test approval/rejection letters still work at
    - http://localhost:3000/api/v2/admin/application-letter/250/rejection/test
    - http://localhost:3000/api/v2/admin/application-letter/250/aproval/test
2. Check that the new yukon grant student template works at 
    - http://localhost:3000/api/v2/admin/application-letter/250/approval/yukon-grant-student
    - localhost:3000/api/v2/admin/application-letter/250/approval/yukon-grant-student.pdf
    - localhost:3000/api/v2/admin/application-letter/250/approval/yukon-grant-student.html
